File: FilmHive/UI/filmHive/models/filmReview.js
import User from "./user";
import Film from "./film";

class FilmReview {
    constructor(data) {
        this.filmReviewId = data.filmReviewId;
        this.userId = data.userId;
        this.movieId = data.movieId;
        this.grade = data.grade;
        this.comment = data.comment;
        this.reviewDate = data.reviewDate;
        this.isDeleted = data.isDeleted;
        this.isActive = data.isActive;
        this.timeOfDeletion = data.timeOfDeletion;
        this.createdAt = data.createdAt;
        this.modifiedAt = data.modifiedAt;
        this.modifiedBy = data.modifiedBy;
        this.user = data.user ? new User(data.user) : null;
        this.movie=data.movie?new Film(data.movie):null;
    }
}

export default FilmReview;

File: FilmHive/filmHive.Services/Database/FilmReview.cs
using System;
using System.Collections.Generic;

namespace filmHive.Services.Database;

public partial class FilmReview: ISoftDelete, ICreated, IModified
{
    public int FilmReviewId { get; set; }

    public int UserId { get; set; }

    public int MovieId { get; set; }

    public decimal? Grade { get; set; }

    public string? Comment { get; set; }

    public DateTime ReviewDate { get; set; }

    public bool IsDeleted { get; set; }

    public bool IsActive { get; set; }

    public DateTime? TimeOfDeletion { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? ModifiedAt { get; set; }

    public int? ModifiedBy { get; set; }

    public virtual Film Movie { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}
